### PR TITLE
Ng 16 save file

### PR DESCRIPTION
**Pull request by alejoreyes96**
Implemented Save file functionality. The UI has a Save File button in which, when the user clicks it, it will prompt a dialog box in which the user has to write a desired file name for their file. If none is written, the method gives it a default name accordingly. 

This save method writes in a file the content of the registers and first 50 locations on memory at the moment of saving, meaning, it can be done at any point in the step-by-step runner.

### CODE REVIEW

File: GUI/window.py [hunk 1]
```diff
@@ -458,12 +475,56 @@ def __init__(self, **kwargs):
         self.add_widget(self.hex_keyboard_label)
         self.add_widget(self.seven_segment_display)
 
-    def save(self, instance):
-        toast("Not Implemented yet. Will be ready on Sprint 3")
+    def open_save_dialog(self, instance):
+        """It will be called when user click on the save file button.
+
+        :param instance: used asevent handler for button click;
```
Comment (ChristianPerez34) [inline GUI/window.py]: :param instance: used as event handler for button click;

File: GUI/window.py [hunk 2]
```diff
@@ -458,12 +475,56 @@ def __init__(self, **kwargs):
...
+    def save_file(self, *args):
+        """It is called when user clicks on 'Save' or 'Cancel' button of dialog.
+
+        :type *args: object array
+        :param *args: passes an object array generated when opening open_save_dialog. 
+                Said object indcludes input text used for file saving;
+
+        """
+        if args[0] == 'Save':
+            filename = ''
```
Comment (ChristianPerez34) [inline GUI/window.py]: filename = 'Save'

File: GUI/window.py [hunk 3]
```diff
@@ -458,12 +475,56 @@ def __init__(self, **kwargs):
...
+
+            # Checks if user input is valid or null for filename. if null, assigns a default filename
+            if args[1].text_field.text:
+                filename = args[1].text_field.text
+            else:
+                filename = 'Save'
```
Comment (ChristianPerez34) [inline GUI/window.py]: Can remove this

**Review by ChristianPerez34 — CHANGES_REQUESTED** [review]
Small changes. Good job.

**Review by ChristianPerez34 — APPROVED** [review]

**Review by noelvalentin — APPROVED** [review]

**Review by cfboy — APPROVED** [review]

**Pull request merged**